.github/workflows/tilegym-ci.yml

```diff
@@ -46,6 +47,20 @@ jobs:
           fi
       - name: Checkout code
         uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Check for code changes
+        id: changed-files
+        uses: step-security/changed-files@v46
+        with:
+          files: |
+            src/**
```

Comment:
If we are going to do this, I think we should modify this to skip if only README/markdown files are updated, rather than hardcoding directories in the repo. I don't want to preclude new directory addition, and it might get confusing if someone thinks they have explicitly enabled tests but nothing is happening